src/lib/stepper/stepper-vertical.html

```diff
@@ -0,0 +1,37 @@
+<div class="mat-stepper-container">
+    <div *ngFor="let step of _steps; let i = index" class="mat-step" role="tablist">
+        <div #stepHeader class="mat-step-header" role="tab"
+             [id]="_getStepLabelId(i)"
+             [attr.aria-controls]="_getStepContentId(i)"
+             [attr.aria-selected]="selectedIndex == i"
+             [tabIndex]="_focusIndex == i ? 0 : -1"
+             (click)="select(step)"
+             (keydown)="_onKeydown($event)">
+            <div *ngIf="step.active" class="active-step">{{i + 1}}</div>
+            <div *ngIf="!step.active" class="inactive-step">{{i + 1}}</div>
+
+            <div class="mat-step-label">
+                <!-- If there is a label template, use it. -->
+                <ng-template [ngIf]="step.stepLabel">
+                    <ng-template [cdkPortalHost]="step.stepLabel"></ng-template>
+                </ng-template>
+                <!-- It there is no label template, fall back to the text label. -->
+                <ng-template [ngIf]="!step.stepLabel">
+                    <div>{{step.label}}</div>
+                </ng-template>
+            </div>
+
+        </div>
+        <div *ngIf="!step._isLast" class="connector-line"></div>
```

Comment:
Is `step._isLast` simply referring to index here? If so, you could use the built-in prop from ngFor on line 2 rather than creating a property for it:

```html
<div *ngFor="let step of _steps; let i = index; let isLast = last" class="mat-step" role="tablist">
...
<div *ngIf="isLast" class="connector-line"></div>
```